### PR TITLE
collision groups for physical objects

### PR DESCRIPTION
**Pull request by AndrewMeadows**
This lays the foundation for using collision groups in the physics engine.  This PR doesn't actually make any visible changes -- objects should still collide (or not, as the case may be) as they used to.  The good news is that it will now be relatively easy to USE the new collision groups to make new features/behavior.

### CODE REVIEW

File: libraries/physics/src/PhysicsEngine.cpp [hunk 1]
```diff
@@ -417,34 +431,7 @@ void PhysicsEngine::setCharacterController(DynamicCharacterController* character
...
-    }
-
-    ObjectMotionState* motionState = static_cast<ObjectMotionState*>(physicsInfo);
-    btRigidBody* body = motionState->getRigidBody();
-    if (!body) {
-        return false;
-    }
-
-    return body->isActive();
-}
```
Comment (AndrewMeadows) [inline libraries/physics/src/PhysicsEngine.cpp]: PhysicsEngine::physicsInfoIsActive() moved to ObjectMotionState::isActive()

File: libraries/physics/src/PhysicsEngine.cpp [hunk 2]
```diff
@@ -417,34 +431,7 @@ void PhysicsEngine::setCharacterController(DynamicCharacterController* character
...
-    btRigidBody* body = motionState->getRigidBody();
-    if (!body) {
-        return false;
-    }
-
-    const btTransform& worldTrans = body->getCenterOfMassTransform();
-    positionReturn = bulletToGLM(worldTrans.getOrigin()) + ObjectMotionState::getWorldOffset();
-    rotationReturn = bulletToGLM(worldTrans.getRotation());
-
-    return true;
```
Comment (AndrewMeadows) [inline libraries/physics/src/PhysicsEngine.cpp]: PhysicsEngine::getBodyLocation() was removed (unused (debug cruft?)).

File: libraries/physics/src/EntityMotionState.cpp [hunk 3]
```diff
@@ -517,3 +518,16 @@ QString EntityMotionState::getName() {
...
+    switch (computeObjectMotionType()){
+        case MOTION_TYPE_STATIC:
+            return COLLISION_GROUP_STATIC;
+        case MOTION_TYPE_KINEMATIC:
+            return COLLISION_GROUP_KINEMATIC;
+        default:
+            break;
+    }
+    return COLLISION_GROUP_DEFAULT;
+}
```
Comment (AndrewMeadows) [inline libraries/physics/src/EntityMotionState.cpp]: At the moment the collision group maps directly to the motion type for entities, but eventually we'll use some more groups for other special cases.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b0ca2df3aa52e4b73242bd8ba173a13c49f910d8 available at: https://gist.github.com/f6bab580a0c679e6cf2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5037.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5037.dmg

**Comment by AndrewMeadows** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: b0ca2df3aa52e4b73242bd8ba173a13c49f910d8 available at: https://gist.github.com/f6bab580a0c679e6cf2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5037.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5037.dmg

**Pull request merged**